core/cap-0034.md

```diff
@@ -118,86 +217,241 @@ older protocols' behavior, and continue to use old behavior until after
+| 6                           | Expect `BASIC` or `SIGNED` `StellarValue`s in ballot protocol?                                      | `BASIC`                                     | `BASIC`                                                     | **`SIGNED`**                                                                 |
 
 ## Design Rationale
 
+It has seemed from the beginning that the problem could be addressed only
+through a CAP: it is a question of what relationships to maintain among the
+payment of fees, the consumption of sequence numbers, and the validity of
+transactions, that also leads to a question of what value to nominate given a
+set of candidates, and all of those questions must be answered by the Core
+through consensus.
+
+1. The proposal prevents the race that motivated the CAP; it ensures in all
+cases that a transaction that returns `txTOO_LATE` does not consume a sequence
+number and does not pay a fee.  This is ensured because before a transaction
+set gets to the point of having its transaction fees paid (which happens when a
+ledger is closing), we combine the candidate `<txSet,closeTime>` pairs into
+one, and with the CAP behavior we do so by simply selecting one candidate.
+And every candidate has by the time of combining been validated, including
+against the new condition that it contains no transactions that are expired
+with respect to its `closeTime`.
+
+2. The proposal additionally repairs a degree of lack of Byzantine fault
+tolerance, already described in the "Motivation" section, which we called
+"maximum `closeTime` injection":  an ill-behaved validator can no longer
+induce the compositing function to produce transaction sets containing
+arbitrarily few transactions _not_ guaranteed to expire before being applied
+by nominating a very large `closeTime`.  As detailed in the "Motivation"
+section, a node would have to provide a transaction set with the best set of
+_unexpired_ transactions in order to influence the composited `closeTime`.
+
+3. This proposal refines [PR 2608](https://github.com/stellar/stellar-core/pull/2608)'s
+trimming of transactions prior to initial nomination by changing the
+next-ledger-`closeTime` estimate (which PR2608 itself had changed from
+last-ledger-`closeTime`) to `closeTime`-within-`StellarValue`.  Given this CAP's
+semantics, we know at this point that that `closeTime` is the _exact_ one that
+will be chosen as the ledger close time _if_ the transaction set wins, so we can
+trim precisely the most efficient set of transactions:  only those which would
+have been guaranteed to fail with `txTOO_LATE`, and none which will fail with
+`txTOO_LATE` if the transaction set is ultimately externalized.
```

Comment:
Either this is contradictory, or I don't understand it. "only those which would have been guaranteed to fail with `txTOO_LATE`, and none which will fail with `txTOO_LATE` if the transaction set is ultimately externalized."

Reply:
I've tried to clarify the wording.